cloud.foundry.cli/src/test/java/cloud/foundry/cli/logic/ApplyLogicTest.java

```diff
@@ -172,193 +169,174 @@ private DefaultCloudFoundryOperations mockSpaceDevelopersGetAll(List<String> spa
-        ApplicationSummary summary3 = createMockApplicationSummary(appManifest3);
-
-        Applications applicationsMock = mock(Applications.class);
-        DefaultCloudFoundryOperations cfOperationsMock = createMockCloudFoundryOperations(
-                                                            Arrays.asList(summary1, summary2, summary3),
-                                                            Arrays.asList(appManifest1, appManifest2, appManifest3),
-                                                            applicationsMock);
-
-        Void voidMock = mock(Void.class);
-        Mono<Void> deletedMonoMock = Mono.just(voidMock);
+        Metadata app3Metadata = createMockMetadata("app3meta", "some/path");
 
-        // This contains all the DeleteApplicationRequests received when delete is called
-        CopyOnWriteArrayList<DeleteApplicationRequest> receivedDeleteRequests = new
-                                                           CopyOnWriteArrayList<DeleteApplicationRequest>();
+        Map<String, ApplicationManifest> apps = new HashMap<String, ApplicationManifest>() {{
+            put("app1", appManifest1);
+            put("app2", appManifest2);
+            put("app3", appManifest3);
+        }};
+        Map<String, Metadata> metadata = new HashMap<String, Metadata>() {{
+            put("app1", app1Metadata);
+            put("app2", app2Metadata);
+            put("app3", app3Metadata);
+        }};
 
-        when(applicationsMock.delete(any(DeleteApplicationRequest.class)))
-            .thenAnswer((Answer<Mono<Void>>) invocation -> {
-                receivedDeleteRequests.add(invocation.getArgument(0));
-
-                return deletedMonoMock;
-            });
+        DefaultCloudFoundryOperations cfOperationsMock = createMockCloudFoundryOperations2(apps, metadata);
 
         ApplyLogic applyLogic = new ApplyLogic(cfOperationsMock);
 
         // when
         applyLogic.applyApplications(appsToApply);
 
         // then
-        verify(applicationsMock).list();
+        DeleteApplicationRequest request1 = DeleteApplicationRequest.builder().name("app2").build();
+        verify(cfOperationsMock.applications(), times(1)).delete(request1);
+        DeleteApplicationRequest request2 = DeleteApplicationRequest.builder().name("app3").build();
+        verify(cfOperationsMock.applications(), times(1)).delete(request2);
+        verify(cfOperationsMock.applications()).list();
+    }
 
-        ListIterator<DeleteApplicationRequest> listDeleteRequests = receivedDeleteRequests.listIterator();
-        assertThat(receivedDeleteRequests.size(), is(2));
 
-        DeleteApplicationRequest deleteRequest1 = listDeleteRequests.next();
-        assertThat(deleteRequest1, is(notNullValue()));
-        assertThat(deleteRequest1.getName(), is("app2"));
+    private ApplicationManifest createMockApplicationManifest(String appName, String path, String buildpack) {
```

Comment:
I would not describe the result of this method as a mock, because it is not a mocked object. We could use the name `createExampleApplicationManifest` instead